### PR TITLE
FIX #57

### LINKED ISSUE #57: Refactor metadata to be a table rather than a string
Currently the metadata is read from a file and returned as a string. Any method that then needs the metadata must create it's own table from it. Instead, in DefaultPointsRenderer, convert this string into a table of key, value pairs and store it. (Note, dont do this in the metadata viewer as converting to a table will throw out duplicate key names). The metadata table (and not string) should then be passed to methods and not the string metadata.

E.g.
```
local metadata = {}
metadata["AF Point"] = "C6"
metadata["Focal Length"] = "30mm"
```

### PR DESCRIPTION
**Pull request by rderimay**
Added a new method ExifUtils.readMetaDataAsTable which return a lua
table with all exif keys and values
Adapted ExfiUtils.findFirstMatchingValue accordingly
Removed unneeded methods

### CODE REVIEW

File: focuspoints.lrdevplugin/ExifUtils.lua [hunk 1]
```diff
@@ -42,37 +42,44 @@ function ExifUtils.readMetaData(targetPhoto)
...
+-- Ignores nil and "(none)" as valid values
+-- metaDataTable - the medaData key/value table
+-- keys - the keys to be search for in order of importance
+--]]
+function ExifUtils.findFirstMatchingValue(metaDataTable, keys)
   local value = nil
 
-  for key,keyword in pairs(keys) do
-    value = ExifUtils.findValue(metaData, keyword)
+  for key, keyword in pairs(keys) do
```
Comment (musselwhizzle) [inline focuspoints.lrdevplugin/ExifUtils.lua]: nit: 
this is typically a "key -> value" loop similar to a HashMap. Mind calling k,v or key, value (and renaming the prop on line 74)?

File: focuspoints.lrdevplugin/ExifUtils.lua [hunk 2]
```diff
@@ -42,37 +42,44 @@ function ExifUtils.readMetaData(targetPhoto)
...
+    value = metaDataTable[keyword]
+    if value == nil then
```
Comment (musselwhizzle) [inline focuspoints.lrdevplugin/ExifUtils.lua]: can we simplify this
```
if value ~= nil and value ~= "(none)" then
   return value
end
```

**Comment by musselwhizzle** [comment]
nice work. Great use of a branch

**Comment by musselwhizzle** [comment]
approved. will give you time to push/respond if you want before i merge.

File: focuspoints.lrdevplugin/ExifUtils.lua (hunk 1 above)
Comment (rderimay) [inline focuspoints.lrdevplugin/ExifUtils.lua]: ok no worries

File: focuspoints.lrdevplugin/ExifUtils.lua (hunk 2 above)
Comment (rderimay) [inline focuspoints.lrdevplugin/ExifUtils.lua]: what do you want me to simplify there ?

Comment (musselwhizzle) [inline focuspoints.lrdevplugin/ExifUtils.lua]: it's fine. you can keep the log if you really want it.

**Comment by musselwhizzle** [comment]
merged. nice work.

**Pull request merged**